magicctapipe/scripts/mars/tests/test_compare_params.py

```diff
@@ -0,0 +1,223 @@
+        "width_M1",
+        "width_M2",
+        "length_M1",
+        "length_M2",
+        "size_M1",
+        "size_M2",
+        "delta_M1",
+        "delta_M2",
+        "cogx_M1",
+        "cogx_M2",
+        "cogy_M1",
+        "cogy_M2",
+        "corex",
+        "corey",
+        "az",
+        "zd",
+        "hmax",
+        "impact_M1",
+        "impact_M2",
+        "slope_M1",
+        "slope_M2",
+    ]
+
+    source = MAGICEventSource(
+        input_url=dataset_calibrated_M1,
+        process_run=False,
+    )
+
+    is_mc = source.is_simulation
+    run_number = source.run_numbers[0]
+
+    if is_mc:
+        config_params = {
+            "magic-cta-pipe-input": {
+                "MCP-path": str(
+                    test_data / "simulated/dl1_stereo" / f"dl1_stereo_magic_only_MAGIC_GA_za35to50.Run{run_number}.h5"
+                )
+            },
+            "MARS-input": {"MARS-path": str(dataset_superstar)},
+            "Params": params_list,
+            "Output_paths": {
+                "file_output_directory": str(test_data / "simulated/test_params"),
+                "image_output_directory": str(test_data / "simulated/test_params"),
+            },
+        }
+    else:
+        config_params = {
+            "magic-cta-pipe-input": {
+                "MCP-path": str(
+                    test_data / "real/dl1_stereo" / f"dl1_stereo_magic_only_MAGIC.Run0{run_number}.h5"
+                )
+            },
+            "MARS-input": {"MARS-path": str(dataset_superstar)},
+            "Params": params_list,
+            "Output_paths": {
+                "file_output_directory": str(test_data / "real/test_params"),
+                "image_output_directory": str(test_data / "real/test_params"),
+            },
+        }
+
+    config_params_file = str(tmp_path / "compare_params_config.yaml")
+
+    with open(config_params_file, "w") as outfile:
+        yaml.dump(config_params, outfile, default_flow_style=False)
+
+    config_mcp = {
+        "mc_tel_ids": {"LST-1": 1, "MAGIC-I": 2, "MAGIC-II": 3},
+        "MAGIC": {
+            "image_extractor": {
+                "type": "SlidingWindowMaxSum",
+                "window_width": 5,
+                "apply_integration_correction": False,
+            },
+            "charge_correction": {"use": True, "correction_factor": 1.143},
+            "magic_clean": {
+                "use_time": True,
+                "use_sum": True,
+                "picture_thresh": 6,
+                "boundary_thresh": 3.5,
+                "max_time_off": 4.5,
+                "max_time_diff": 1.5,
+                "find_hotpixels": "auto",
+                "pedestal_type": "from_extractor_rndm",
+            },  # select 'fundamental', 'from_extractor' or 'from_extractor_rndm'
+            "muon_ring": {
+                "thr_low": 25,
+                "tailcut": [12, 8],
+                "ring_completeness_threshold": 25,
+            },
+        },
+        "event_coincidence": {
```

Comment:
this part should be only needed for MAGIC+LST1 joining,
MAGIC alone is joined by stereo event number